client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
+	const settings = {};
+	settings[ USER_SETTING_KEY ] = getState().preferences.values;
+	wpcom.me().settings().update( JSON.stringify( settings ) ).then( ( data ) => {
+		dispatch( {
+			type: PREFERENCES_RECEIVE,
+			data
+		} );
+	} );
+};
+
+export const setPreference = ( key, value ) => ( dispatch, getState ) => {
+	dispatch( {
+		type: PREFERENCES_SET,
+		key,
+		value
+	} );
+	save()( dispatch, getState );
+};
+
+export const removePreference = ( key ) => ( dispatch, getState ) => {
```

Comment:
I'd say there's a reasonable amount of certainty we're not using `remove` anywhere in Calypso:

```
$ grep -r -i preferences/actions client
client/components/data/preferences-data/index.jsx:var PreferencesActions = require( 'lib/preferences/actions' ),
client/components/tinymce/plugins/advanced/plugin.jsx:import PreferencesActions from 'lib/preferences/actions';
client/components/tinymce/plugins/after-the-deadline/plugin.js:import PreferencesActions from 'lib/preferences/actions';
client/layout/community-translator/invitation-utils.js:	preferencesActions = require( 'lib/preferences/actions' ),
client/layout/poll-invitation/index.js:import preferencesActions from 'lib/preferences/actions';
client/lib/preferences/README.md:var PreferencesActions = require( 'lib/preferences/actions' );
client/lib/sites-list/list.js:	PreferencesActions = require( 'lib/preferences/actions' ),
client/my-sites/media-library/content.jsx:import PreferencesActions from 'lib/preferences/actions';
client/post-editor/media-modal/gallery-help-container.jsx:import PreferencesActions from 'lib/preferences/actions';
client/post-editor/post-editor.jsx:	PreferencesActions = require( 'lib/preferences/actions' ),
client/post-editor/test/post-editor.jsx:		mockery.registerMock( 'lib/preferences/actions', { set() {} } );
```

```js
$ grep -r -i preferencesactions.remove client
client/lib/preferences/actions.js:PreferencesActions.remove = function( key ) {
client/lib/preferences/test/actions.js:			PreferencesActions.remove( 'one' );
```

YAGNI